projects/mysql-cluster/README.md

```diff
@@ -0,0 +1,218 @@
+         MASTER_LOG_POS=<position_from_master_status>;
+     START SLAVE;
+     ```
+
+5. **Verify Replication**
+   - Check the replica status:
+     ```sql
+     SHOW REPLICA STATUS\G;
+     ```
+   - Ensure `Slave_IO_Running` and `Slave_SQL_Running` are both `Yes`.
+   - Something isn't quite right. Can you figure out how to fix this? There may be a few things that need to be fixed. Use `SHOW REPLICA STATUS\G` to find out what is wrong. Can you see the `cyfdb` database on the replica?
+
+### Task 3: Demonstrate Replication
+
+1. **Insert Data into the Primary Server**
+   - On the primary server, insert a new record:
+     ```sql
+     INSERT INTO cyfdb.users (name, email) VALUES ('John Doe', 'john@example.com');
+     ```
+
+2. **Verify Data on the Secondary Server**
+   - On the secondary server, query the table:
+     ```sql
+     SELECT * FROM testdb.users;
+     ```
+   - Verify that the data matches the primary server.
+   - Execute the following on the secondary server:
+   ```sql
+   INSERT INTO cyfdb.users (name, email) VALUES ('Jane Doe', 'jane@example.com');
+   ```
+   Did it work? If not, can you guess why?
+
+### Task 4: Demonstrate Failover
+
+1. **Simulate Primary Server Failure**
+   - Stop the MySQL service on the primary server:
+     ```bash
+     sudo systemctl stop mysql
+     ```
+
+2. **Promote Secondary Server to Primary**
+   - On the secondary server, stop the replica:
+     ```sql
+     STOP REPLICA;
+     ```
+   - Reset the replica configuration:
+     ```sql
+     RESET REPLICA ALL;
+     ```
+   - Ensure the secondary server can accept writes by setting the read-only mode to off:
+     ```sql
+     SET GLOBAL read_only = OFF;
+     ```
+
+3. **Test Write Operations on the New Primary**
+   - Insert new data into the secondary server (now acting as the primary):
+     ```sql
+     INSERT INTO cyfdb.users (name, email) VALUES ('Jane Doe', 'jane@example.com');
+     ```
+   - Query the table to ensure the new data is inserted:
+     ```sql
+     SELECT * FROM cyfd.users;
+     ```
+
+### Task 5: Reconfigure Original Primary as Secondary
+
+1. **Reconfigure the Original Primary**
+   - Start the MySQL service on the original primary:
+     ```bash
+     sudo systemctl start mysql
+     ```
+   - On the original primary server, set it up as the slave to the new primary:
+     ```sql
+     CHANGE MASTER TO
+         MASTER_HOST='<new_primary_ip>',
+         MASTER_USER='replica_user',
+         MASTER_PASSWORD='C90L6`!Doe{K',
+         MASTER_LOG_FILE='<file_name_from_new_primary>',
+         MASTER_LOG_POS=<position_from_new_primary>;
+     START SLAVE;
+     ```
+   - Check the replica status:
+     ```sql
+     SHOW REPLICA STATUS\G;
+     ```
+
+2. **Verify Data Synchronization**
+   - Insert new data on the new primary and verify it replicates to the original primary.
+
+   This step may again not work straight away. Refer to how you configured the original primary, did you follow every step for this server?
```

Comment:
```suggestion
   This step may again not work straight away. Refer to how you configured the original primary: did you follow every step for this server?
```

Get some more juice out of this by getting them to write down a prediction first and then suggest how to test the prediction (by inserting new data)

Reply:
This isn't guaranteed to fail. The steps don't mandate doing some of the config on this server, but they might still do them in which case it will work perfectly.